plugins/hdfsviewer/unit/azkaban/viewer/hdfs/ORCFileViewerTest.java

```diff
@@ -0,0 +1,206 @@
+    }
+
+    @After
+    public void tearDown() throws IOException {
+        fs.close();
+    }
+
+    /* Calls ORCFileViewer#displayFile and parse results */
+    String displayRecordWrapper(String filename, int startRecord, int endRecord)
+        throws IOException {
+        ByteArrayOutputStream outStream = new ByteArrayOutputStream();
+        viewer.displayFile(fs, getResourcePath(filename), outStream,
+            startRecord, endRecord);
+        String records = new String(outStream.toByteArray());
+        records = records.replaceAll("Record [0-9]*:", "");
+        records = StringUtils.deleteWhitespace(records);
+        return records;
+    }
+
+    /* Get Path to a file from resource dir */
+    Path getResourcePath(String filename) {
+        URL url =
+            Thread.currentThread().getContextClassLoader()
+                .getResource("resources/" + filename);
+        return new Path(url.getPath());
+    }
+
+    /* verify capability for empty orc files */
+    @Test
+    public void orcEmptyFileCapability() throws IOException {
+        assertEquals(supportedCapabilities, viewer.getCapabilities(fs,
+            getResourcePath("TestOrcFile.emptyFile.orc")));
+    }
+
+    /* verify capability for generic orc files */
+    @Test
+    public void genericORCFileCapability() throws IOException {
+        assertEquals(supportedCapabilities, viewer.getCapabilities(fs,
+            getResourcePath("TestOrcFile.testPredicatePushdown.orc")));
+    }
+
+    /* verify capability for orc files with binary type */
+    @Test
+    public void binaryTypeORCFileCapability() throws IOException {
+        assertEquals(supportedCapabilities, viewer.getCapabilities(fs,
+            getResourcePath("TestOrcFile.testStringAndBinaryStatistics.orc")));
+    }
+
+    /* verify capability for snappy compressed orc file */
+    @Test
+    public void snappyCompressedORCFileCapability() throws IOException {
+        assertEquals(supportedCapabilities, viewer.getCapabilities(fs,
+            getResourcePath("TestOrcFile.testSnappy.orc")));
+    }
+
+    /* verify capability for union type orc file */
+    @Test
+    public void unionTypeORCFileCapability() throws IOException {
+        assertEquals(supportedCapabilities, viewer.getCapabilities(fs,
+            getResourcePath("TestOrcFile.testUnionAndTimestamp.orc")));
+    }
+
+    /* verify capability for avro files */
+    @Test
+    public void avroCapability() throws IOException {
+        assertEquals(unSupportedCapabilities,
+            viewer.getCapabilities(fs, getResourcePath("TestAvro.avro")));
+    }
+
+    /* verify capability for text files */
+    @Test
+    public void textCapability() throws IOException {
+        assertEquals(unSupportedCapabilities,
+            viewer.getCapabilities(fs, getResourcePath("TestTextFile.txt")));
+    }
+
+    /* verify capability for parquet files */
+    @Test
+    public void parquetCapability() throws IOException {
```

Comment:
Then maybe we can rename the test to like "noParquetCapability" to reduce confusion?